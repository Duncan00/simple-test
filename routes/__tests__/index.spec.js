'use strict';

const supertest = require('supertest');
const nock = require('nock');
const app = require('../../app');

describe('GET /', () => {
	afterEach(async () => {
		nock.cleanAll();
	});

	describe('When third party API responses 200', () => {
		it('should response OK', async () => {
			const scope = nock('https://www.google.com')
				.get('/')
				.once()
				.reply(200);

			const response = await supertest(app).get('/');

			scope.done();
			expect(response.status).toEqual(200);
			expect(response.body).toMatchSnapshot('200 OK');
		});
	});

	describe('When third party API responses 404', () => {
		it('should response message Not Found', async () => {
			const scope = nock('https://www.google.com')
				.get('/')
				.reply(404);

			const response = await supertest(app).get('/');

			// Assert that the expected request was made.
			scope.done();
			expect(response.status).toEqual(404);
			expect(response.body).toMatchSnapshot('404 Not Found');
		});
	});
});
