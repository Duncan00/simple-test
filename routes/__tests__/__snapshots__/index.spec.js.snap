// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / When third party API responses 200 should response OK: 200 OK 1`] = `
Object {
  "message": "OK",
}
`;

exports[`GET / When third party API responses 404 should response message Not Found: 404 Not Found 1`] = `
Object {
  "message": "Not Found",
}
`;
